### PR TITLE
Add a follow-up request page

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR adds a page to display follow-up requests.

### CODE REVIEW

File: skyportal/tests/api/test_followup_requests_api.py [hunk 1]
```diff
@@ -162,3 +164,56 @@ def test_group1_user_cannot_see_group2_followup_request(
...
+    time_before_post = datetime.utcnow().isoformat()
+    status, data = api(
+        'POST', 'followup_request', data=request_data, token=upload_data_token
+    )
+    assert status == 200
+    assert data['status'] == 'success'
+
+    status, data = api(
+        "GET",
+        f"followup_request?startDate={time_before_post}",
```
Comment (guynir42) [inline skyportal/tests/api/test_followup_requests_api.py:195]: better to put the query arguments in a `params` named argument.

File: skyportal/tests/api/test_followup_requests_api.py [hunk 2]
```diff
@@ -162,3 +164,56 @@ def test_group1_user_cannot_see_group2_followup_request(
...
+        token=view_only_token,
+    )
+    assert status == 200
+    assert data['status'] == 'success'
+    assert any([s['obj_id'] == public_source.id for s in data["data"]])
+
+    time_after_post = datetime.utcnow().isoformat()
+    status, data = api(
+        "GET",
+        f"followup_request?startDate={time_after_post}",
```
Comment (guynir42) [inline skyportal/tests/api/test_followup_requests_api.py:205]: same here.

File: skyportal/tests/api/test_followup_requests_api.py [hunk 3]
```diff
@@ -162,3 +164,56 @@ def test_group1_user_cannot_see_group2_followup_request(
...
+        token=view_only_token,
+    )
+    assert status == 200
+    assert data['status'] == 'success'
+    assert not any([s['obj_id'] == public_source.id for s in data["data"]])
+
+    status, data = api(
+        "GET",
+        f"followup_request?sourceID={public_source.id}",
```
Comment (guynir42) [inline skyportal/tests/api/test_followup_requests_api.py:214]: same here.

File: skyportal/tests/flaky/test_frontend_flaky.py [hunk 4]
```diff
@@ -355,3 +355,62 @@ def test_gcnevents_observations(
     driver.wait_for_xpath('//*[text()="ztfr"]')
     driver.wait_for_xpath('//*[text()="1.57415"]')
     driver.wait_for_xpath('//*[text()="20.40705"]')
+
+
+# @pytest.mark.flaky(reruns=2)
```
Comment (guynir42) [inline skyportal/tests/flaky/test_frontend_flaky.py:360]: I'm worried that we are putting all the frontend tests directly into the flaky category. Also there's a commented line here that should probably not be.

File: static/js/components/FollowupRequestSelectionForm.jsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    await dispatch(fetchFollowupRequests(formData));
+    setIsSubmitting(false);
+  };
+
+  function validate(formData, errors) {
+    if (formData.start_date > formData.end_date) {
+      errors.start_date.addError(
+        "Start date must be before end date, please fix."
+      );
+    }
```
Comment (guynir42) [inline static/js/components/FollowupRequestSelectionForm.jsx:22]: Are there no other restrictions on time (past or future times etc)?

**Review by guynir42 — COMMENTED** [review]
Minor comments, but looks good overall.

File: skyportal/tests/api/test_followup_requests_api.py (hunk 1 above)
Comment (mcoughlin) [reply skyportal/tests/api/test_followup_requests_api.py]: fixed.

File: skyportal/tests/api/test_followup_requests_api.py (hunk 2 above)
Comment (mcoughlin) [reply skyportal/tests/api/test_followup_requests_api.py]: fixed.

File: skyportal/tests/api/test_followup_requests_api.py (hunk 3 above)
Comment (mcoughlin) [reply skyportal/tests/api/test_followup_requests_api.py]: fixed.

File: skyportal/tests/flaky/test_frontend_flaky.py (hunk 4 above)
Comment (mcoughlin) [reply skyportal/tests/flaky/test_frontend_flaky.py]: I uncommented it. I am mostly worried about the tradeoff here between something breaking and these elements interrupting others. The less complicated ones I am still writing into frontend/

File: static/js/components/FollowupRequestSelectionForm.jsx (hunk 5 above)
Comment (mcoughlin) [reply static/js/components/FollowupRequestSelectionForm.jsx:22]: Not really I think, the requests span windows of time.

**Comment by mcoughlin** [comment]
@guynir42 Please merge if happy.

File: skyportal/tests/api/test_followup_requests_api.py [hunk 6]
```diff
@@ -190,28 +190,38 @@ def test_filter_followup_request(
     assert status == 200
     assert data['status'] == 'success'
 
+    params = {'startDate': time_before_post}
```
Comment (guynir42) [inline skyportal/tests/api/test_followup_requests_api.py:193]: FYI you can put the params dictionary directly inline if it is not too long.

**Review by guynir42 — APPROVED** [review]

**Pull request merged**